### ISSUE TITLE
Updated documentation or Wiki

### ISSUE THREAD

**Issue by julianjupiter** [eclipse-ee4j/glassfish-hk2#514]
The website, https://javaee.github.io/hk2/, remains temporarily paused until now. I'm struggling looking for the updated documentation of **HK2** (as well as **Grizzly**). The link of status provided returns 404: https://www.eclipse.org/ee4j/status.php.
![hk2-01](https://user-images.githubusercontent.com/15760709/85065422-b3c3d300-b1df-11ea-8481-5ea52f1e44d7.JPG)
![hk2-02](https://user-images.githubusercontent.com/15760709/85065426-b4f50000-b1df-11ea-9ae7-e79ca83a357f.JPG)

**Comment by arjantijms** [comment]
At the moment I don't think there's anyone really working on this. Maybe @Cousjava would be interested? If you like to help out yourself too together with Jonathan I'm sure that would be much appreciated.

**Comment by smillidge** [comment]
I don't think the documentation was donated

**Comment by julianjupiter** [comment]
> I don't think the documentation was donated

Does that mean a new documentation has to be created from scratch?

**Comment by CrazyBolillo** [comment]
Is there any outline of things the new documentation should cover?

**Comment by jwells131313** [comment]
Can you go back to Oracle and ask for rights to the doc?  I wrote most of it but it's owned by Oracle.  I really don't think they would mind donating it if it would help get this kick started

**Comment by smillidge** [comment]
@jwells131313 I can but would it be better for you to reach out to @edbratt or @m0mus or someone like that on the Oracle side?

**Comment by rkraneis** [comment]
Did someone find time to look into this? As @julianjupiter mentions, it's quite hard to find current documentation (or ask for clarification on the current one).

**Comment by smillidge** [comment]
It needs someone on the Oracle side to see if docs can be donated to the Eclipse Foundation.

**Comment by rkraneis** [comment]
Hmm, are they even aware? From following the Apache NetBeans donation as a bystander I got the impression that sometimes some shoving is necessary ... :thinking:

**Comment by julianjupiter** [comment]
Right. The initial action should come outside Oracle, not the other way
around. So someone should ask Oracle. Let’s not wait for someone from
Oracle.

**Comment by arjantijms** [comment]
@julianjupiter indeed, can you contact Oracle? Maybe start with @m0mus or @edbratt ?

**Comment by smillidge** [comment]
I'll raise at the PMC meeting Grizzly and HK2 documentation

**Comment by edbratt** [comment]
Generally, this was always planned, but (as with many issues like this) became overrun with other seemingly more urgent tasks and never reached the threshold of actual action. We can package up the material that was available in the javaee github organization at the time of contribution -- wash and tidy up the licenses and copyrights to meet Eclipse requirements -- and submit it for IP review (which should be rather perfunctory) -- then check it into the Eclipse GH repository.
Just to be clear - we'd be contributing the contents of [java ee hk2 -- gh-pages branch](https://github.com/javaee/hk2/tree/gh-pages). From here, someone would need to make any necessary updates to reflect any of the latest changes. 
[I'm not sure it it makes sense to contribute the JavaDocs. Probably better to just use the current version(s) -- comments on this?]
Would that be sufficient (you can look at the legacy site to see what the contribution will consist of [hk2-site](https://javaee.github.io/hk2/) ([grizzly-site](https://javaee.github.io/grizzly/)))? 
[I can also update that "status" link since the page it used to point to seems to be gone :-( ]

**Comment by smillidge** [comment]
I think the gh-pages is sufficient (assuming those docs sites are generated from the gh-pages)

**Comment by edbratt** [comment]
I've put a prospective Doc. update from my personal GH workspace ([here](https://github.com/edbratt/hk2-site-test)). The current GH Page view is [here](https://edbratt.github.io/hk2-site-test/introduction). I'm waiting on feedback from the Eclipse EMO about the contribution process. Once that's cleared up, I can submit these for the HK2 GH Pages branch.

**Comment by edbratt** [comment]
I have committed, on the [gh-pages branch](https://github.com/eclipse-ee4j/glassfish-hk2/commit/73ede92a397102a3375d95affe24a25dcdcc588b) Markdown content that was available from the legacy javaee/hk2 repository. I had intended to submit these changes through a PR, but I confused myself between my fork and the upstream repository and decided it wasn't worth trying to back it out. Sorry, but the content is now available for community use and evolution. 

I will file a ticket for the Eclipse admins to enable the site. The link will be: eclipse-ee4j.github.io/glassfish-hk2.
Next thing will be to update the site content with the current JavaDocs and add content for the current release.

**Comment by edbratt** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=572051

**Comment by edbratt** [comment]
Docs are committed to the [gh-pages branch](https://github.com/eclipse-ee4j/glassfish-hk2/tree/gh-pages) and the [new doc-site ](https://eclipse-ee4j.github.io/glassfish-hk2/)is operating as expected. I'm closing this issue. Further work to enhance/update these docs. can be handled in separate issue(s) and/or PRs. The first to things that come to mind (I've left in the branch readme.md) are add javadocs for the two Eclipse releases and update the docs themselves for v3.0. Hope this is useful.
